runhouse/servers/http/http_server.py

```diff
@@ -308,31 +308,23 @@ def update_settings(request: Request, message: ServerSettings) -> Response:
     @staticmethod
     @app.post("/resource")
     @validate_cluster_access
-    def put_resource(request: Request, message: Message):
-        # if resource is env and not yet a servlet, construct env servlet
-        if message.env and message.env not in env_servlets.keys():
-            resource = b64_unpickle(message.data)[0]
-            if resource["resource_type"] == "env":
-                runtime_env = (
-                    {"conda_env": resource["env_name"]}
-                    if resource["resource_subtype"] == "CondaEnv"
-                    else {}
-                )
-
-                _ = ObjStore.get_env_servlet(
-                    env_name=message.env,
-                    create=True,
-                    runtime_env=runtime_env,
-                    resources=resource.get("compute", None),
-                )
-
-        return HTTPServer.call_in_env_servlet(
-            "put_resource",
-            [message],
-            env=message.env,
-            create=True,
-            lookup_env_for_name=message.key,
-        )
+    def put_resource(request: Request, params: PutResourceParams):
+        HTTPServer.register_activity()
+        try:
+            env_name = params.env_name or "base"
+            return obj_store.put_resource(
+                serialized_data=params.serialized_data,
+                serialization=params.serialization,
+                env_name=env_name,
+            )
+        except Exception as e:
+            logger.exception(e)
+            HTTPServer.register_activity()
+            return Response(
+                error=pickle_b64(e),
+                traceback=pickle_b64(traceback.format_exc()),
```

Comment:
I continue to wonder out loud if we need to serialize these. Most likely not. Strings are probably fine.

Reply:
In the HTTPServer specifically. I don't know for the EnvServlet and obj store.

Reply:
I can remove this later